app/src/features/surveys/view/survey-animals/form-sections/MeasurementAnimalForm.tsx

```diff
@@ -98,23 +86,31 @@ interface MeasurementFormContentProps {
   measurements?: IMeasurementStub[];
 }
 
-const MeasurementFormContent = ({ index, measurements }: MeasurementFormContentProps) => {
+export const MeasurementFormContent = ({ index, measurements }: MeasurementFormContentProps) => {
   const { values, handleChange, setFieldValue, handleBlur } = useFormikContext<IAnimal>();
-  const [measurement, setMeasurement] = useState<IMeasurementStub>();
-
+  const { animalKeyName } = ANIMAL_SECTIONS_FORM_MAP[SurveyAnimalsI18N.animalMeasurementTitle];
   const taxonMeasurementId = values.measurements[index].taxon_measurement_id;
+  const [measurement, setMeasurement] = useState<IMeasurementStub | undefined>(
+    measurements?.find((m) => m.taxon_measurement_id === taxonMeasurementId)
+  );
   const isQuantMeasurement = has(measurement, 'unit');
 
-  const tMeasurementIDName = getAnimalFieldName<IAnimalMeasurement>(NAME, 'taxon_measurement_id', index);
-  const valueName = getAnimalFieldName<IAnimalMeasurement>(NAME, 'value', index);
-  const optionName = getAnimalFieldName<IAnimalMeasurement>(NAME, 'qualitative_option_id', index);
+  const tMeasurementIDName = getAnimalFieldName<IAnimalMeasurement>(animalKeyName, 'taxon_measurement_id', index);
```

Comment:
avoid abbreviations that obfuscate the purpose of a variable (`tMeasurement` could be "time measurement", "telemetry measurement", ...)

Reply:
fixed